ecs-cli/modules/utils/compose/ecs_params_reader.go

```diff
@@ -104,6 +105,49 @@ type AwsVpcConfiguration struct {
 
 type AssignPublicIp string
 
+// ServiceDiscovery hold information related to ECS/Route53 Service Discovery
+type ServiceDiscovery struct {
+	ContainerName           string                  `yaml:"container_name"`
+	ContainerPort           *int64                  `yaml:"container_port"`
+	PrivateDNSNamespace     PrivateDNSNamespace     `yaml:"private_dns_namespace"`
+	PublicDNSNamespace      PublicDNSNamespace      `yaml:"public_dns_namespace"`
+	ServiceDiscoveryService ServiceDiscoveryService `yaml:"service_discovery_service"`
+}
+
+// PrivateDNSNamespace holds information related to Route53 private DNS namespaces
+type PrivateDNSNamespace struct {
+	VPC         string `yaml:"vpc"`
+	ID          string `yaml:"id"`
+	Name        string `yaml:"name"`
+	Description string `yaml:"description"`
+}
+
+// PublicDNSNamespace holds information related to Route53 public DNS namespaces
+type PublicDNSNamespace struct {
+	ID   string `yaml:"id"`
+	Name string `yaml:"name"`
+}
```

Comment:
(No-op musings on encapsulation): Not sure how go-yaml handles embedding, but I wonder if it's possible to use composition for the DNSNamespace types since there's some repetition...

Reply:
I don't think there's enough re-use to justify that abstraction, just figured I'd bring it up as food for thought.